File: det.py
def determinant(matrix):
    if len(matrix) != 2 or len(matrix[0]) != 2 or len(matrix[1]) != 2:
        return "Error: Input matrix must be a 2x2 matrix."
    
    a = matrix[0][0]
    b = matrix[0][1]
    c = matrix[1][0]
    d = matrix[1][1]
    
    return a*d - b*c

# Example usage:
matrix = [
    [2, 3],
    [4, 5]
]

det = determinant(matrix)
print("Determinant:", det)

File: det.m
% Prompt the user to input the 2x2 matrix
matrix = input("Enter the 2x2 matrix [a, b; c, d]: ");

% Check if the input is a 2x2 matrix
if ismatrix(matrix) && all(size(matrix) == [2, 2])
    % Extract elements of the matrix
    a = matrix(1, 1);
    b = matrix(1, 2);
    c = matrix(2, 1);
    d = matrix(2, 2);
    
    % Calculate the determinant
    determinant = a*d - b*c;
    
    % Print the determinant
    fprintf("The determinant of the matrix is: %f\n", determinant);
else
    % Print error message for non-2x2 matrix input
    fprintf("Error: Input must be a 2x2 matrix.\n");
end